custom-metrics-stackdriver-adapter/adapter.go

```diff
@@ -33,9 +36,18 @@ import (
 	coreadapter "github.com/GoogleCloudPlatform/k8s-stackdriver/custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider"
 	adapter "github.com/GoogleCloudPlatform/k8s-stackdriver/custom-metrics-stackdriver-adapter/pkg/adapter/provider"
 	basecmd "github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/cmd"
-	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/provider"
 )
 
+type clock interface {
```

Comment:
You don't need clock and realClock  structs here. 
declare them in query_builder.go